explore/src/main/scala/explore/targets/TargetSelectionTable.scala

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2016-2021 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package explore.targets
+
+import cats.syntax.all._
+import crystal.react.reuse._
+import explore.components.ui.ExploreStyles
+import explore.model.reusability._
+import japgolly.scalajs.react._
+import japgolly.scalajs.react.vdom.html_<^._
+import lucuma.core.model.Target
+import react.common.ReactFnProps
+import react.common.implicits._
+import react.common.style.Css
+import react.semanticui.collections.table.Table
+import react.semanticui.collections.table.TableCell
+import react.semanticui.collections.table.TableCompact
+import react.semanticui.collections.table.TableHeaderCell
+import react.semanticui.collections.table.TableRow
+import react.semanticui.elements.button.Button
+import react.semanticui.sizes
+import reactST.reactTable._
+
+import scalajs.js.JSConverters._
+
+final case class TargetSelectionTable(
+  targets:       List[Target],
+  onSelected:    Target ==> Callback,
+  selectedIndex: Option[Int],
+  onClick:       (Target, Int) ==> Callback
+) extends ReactFnProps[TargetSelectionTable](TargetSelectionTable.component)
+
+object TargetSelectionTable {
+  type Props = TargetSelectionTable
+
+  implicit protected val propsReuse: Reusability[Props] = Reusability.derive
+
+  protected val TargetTable = TableDef[Target].withSortBy
+
+  protected val TargetTableComponent = new SUITable(TargetTable)
+
+  private val columnClasses: Map[String, Css] = Map(
+    "type" -> (ExploreStyles.Sticky |+| ExploreStyles.TargetSummaryType),
+    "name" -> (ExploreStyles.Sticky |+| ExploreStyles.TargetSummaryName)
+  )
+
+  protected val component = ScalaFnComponent
+    .withHooks[Props]
+    // cols
+    .useMemoBy(_ => ()) { props => _ =>
+      List(
```

Comment:
Didn't know you could memoize react fragments

Reply:
I think we can memoize anything, but in this case it's actually a list of table columns.